valhalla/baldr/directededge.h

```diff
@@ -1076,6 +1087,15 @@ class DirectedEdge {
     return superseded_;
   }
 
+  /**
+   * Unlike superseded(), this does not return the raw mask but the shortcut index
+   * that was originally passed to set_superseded().
+   * @return  Returns the index of the set bit in the superseded mask.
+   */
+  uint32_t superseded_idx() const {
+    return log2(superseded_) + 1;
```

Comment:
`log2` does some floating point stuff because it assumes we are trying to do a purely mathematical operation here. what we actually want to know is the position of the first non zero bit. i've never used it before but it seems a function called `ffs` (find first set) does this for us and usually has hardware optimization in modern architectures

Reply:
Good point, I found a few builtins but did not come across `ffs`.

Reply:
Apparently this is not available on Windows, there is this [port of strings.h](https://github.com/win32ports/strings_h) or we implement it ourselves?

Reply:
i had wondered about windows but didnt search for it. that sucks.. and also its in c++20 but we aren't there yet... how annoying. yeah i guess i would use strings.h in a "not windows" ifdef and then down in the function in an ifdef for when it is windows just implement the while loop and put a TODO that we'll use `std::countr_zero + 1` when we move to c++20. what do you prefer here?